engine_test.go

```diff
@@ -15,53 +15,73 @@
 package engine
 
 import (
+	"context"
+	"errors"
+	"path/filepath"
 	"testing"
-)
 
-var TestUnitType UnitType = 999
+	"github.com/stretchr/testify/assert"
+)
 
-type TestRule struct{}
+type successTestRule struct{}
 
-func (rule TestRule) IsFor(unitType UnitType) bool {
-	return TestUnitType == unitType
+func (rule *successTestRule) Run(_ string) ([]Finding, error) {
+	return nil, nil
```

Comment:
Can we change this to return an Finding mock and check that the length of  findings returned from `engine.Run` is equl here?  With this we assert that all concurrency stuffs works correctly and all findings was successful returned

Reply:
Done!